burn-core/src/nn/conv/conv2d.rs

```diff
@@ -74,24 +75,21 @@ pub struct Conv2d<B: Backend> {
 impl Conv2dConfig {
     /// Initialize a new [conv2d](Conv2d) module.
     pub fn init<B: Backend>(&self) -> Conv2d<B> {
-        let k = (self.channels[0] * self.kernel_size[0] * self.kernel_size[1]) as f64;
-        let k = sqrt(1.0 / k);
-
-        let initializer = if let Initializer::UniformDefault = self.initializer {
-            Initializer::Uniform(-k, k)
-        } else {
-            self.initializer.clone()
-        };
-
-        let weight = initializer.init([
+        let shape = Shape::from([
             self.channels[1],
             self.channels[0],
             self.kernel_size[0],
             self.kernel_size[1],
         ]);
-
+        let fan_in = shape.fan_in();
+        let weight = self
+            .initializer
+            .init(shape, InitializerOptions::default().with_fan_in(fan_in));
```

Comment:
I was really trying to come up with something that won't force us to have None arguments on every simple call like
Initializer::Zeros.init(None, None)

Reply:
Yeah I like that

Reply:
I think it's better to have an option struct here, since we may add more arguments in the future and adding an optional field to a struct isn't a breaking change, but adding a new argument to each `init` call is.

Reply:
What about init and init_with, where the latter takes an option struct

Reply:
Looking at the PyTorch init implementation and thinking realistically we will not be adding more options. The options from the struct are passed from the caller to init and they aren't any besides fan in and out. Any other options are part of the enum struct. The true name for the option is actually InitParams, which is small in its scope. We could also refactor if needed but currently we do not have a good use case for the extra flexibility. Lets keep it simple =)